### ISSUE TITLE
Autotile bitmask not working as expected

### ISSUE THREAD

**Issue by eastrd** [godotengine/godot#41320]
**Godot version:**
3.2.2

**OS/device including version:**
Windows 10

**Issue description:**

- A standard `blob` tilesheet that was exported as Godot `.tres` by Tilesetter with bitmasks on didn't work as expected.
- I have also tried to export the tilesheet as PNG and manually draw bitmasks on top, resulted in the same weird behaviour.
- I've tried reimporting the image/`.tres` and recreating the Tilemap node, but all resulted in the same strange behaviour
- Restarting Godot fixed this issue, which to me seems like a bug.
- A screenshot is attached below, hopefully that should clarify the issue (looks like the bitmasks are flipped for some reason):
![Screenshot](https://media.discordapp.net/attachments/212250894228652034/744814857001631774/unknown.png?width=1339&height=702)

**Steps to reproduce:**
Unfortunately, the issue was resolved by restarting Godot, I cannot reproduce the bug.

**Comment by TheDuriel** [comment]
Live editing the tileset may cause tilemaps using it to get a bit confused. Closing and opening the scene, or just toggling visibility, should work fine to solve this.

**Comment by bojidar-bg** [comment]
The image looks like you rotated the tile when placing it. Unfortunately, the autotile code does not support this yet.

**Comment by eastrd** [comment]
> 
> 
> Live editing the tileset may cause tilemaps using it to get a bit confused. Closing and opening the scene, or just toggling visibility, should work fine to solve this.

Thank you TheDuriel, it was fixed after restarting Godot itself.




> 
> 
> The image looks like you rotated the tile when placing it. Unfortunately, the autotile code does not support this yet.

Nope, I didn't rotate the tiles at all. They were placed as shown which I was finding it weird.

**Comment by eastrd** [comment]
Hi, I've just encountered this issue again.
Some autotiles were connecting and some weren't.
I can confirm that closing+reopening the scene does not work. (Forgot to try toggling visibility, will do if it happens again).
And once again restarting Godot fixed it.

**Comment by NathanLovato** [comment]
Is this a docs issue or an issue with the tilesets? To know if we should move it to the docs repo. Hopefully when groud starts working on editor UX he can address issues related to tileset usage.

**Comment by RegineerCZ** [comment]
I have similliar problem, I have this blob for autotile 
![Snímka obrazovky z 2020-11-30 19-31-19](https://user-images.githubusercontent.com/20363297/100649489-cf143c00-3342-11eb-8201-05d4f8d4f39a.png)
And when I draw lines it all looks ok and it’s autotiling as expected.
![Snímka obrazovky z 2020-11-30 19-31-34](https://user-images.githubusercontent.com/20363297/100649537-e18e7580-3342-11eb-80d7-481715bef753.png)
But as soon as I make a turn, the tiles next to the turn change into the "default / icon" tile I chose for this autotile. (If I change icon tile these 3 tiles change accordingly)
![Snímka obrazovky z 2020-11-30 19-31-42](https://user-images.githubusercontent.com/20363297/100649652-0aaf0600-3343-11eb-9dc5-19d5a8e531e8.png)


This is happening in the 3.2.3 from the website, on linux PoPOS!

**Comment by YavorK** [comment]
It looks like I was just having the exact same problem. Tired Saving all scenes, Reloading the scene, togging TileMap on and off and nothing was working. Restarded Godot - tiles were wrong.

Restarting the PC however did work. I had to delete the tiles I added while it was buggy, but after the restart adding NEW tiles worked flawlessly.

**Comment by rpcesar** [comment]
I just wanted to throw out there that I encountered a strikingly similar issue using a 2x2 autotile (3.2.3.stable). At some random point in time (hour or usage maybe?) the editor stopped using auto tile correctly, the bitmask was wrong somehow. The contents of the tileset were still correct. shutting down and restarting the scene did not fix the error. Shutting down and restarting godot entirely DID fix the error. Any tiles drawn while flipped needed to be redrawn.

**Comment by mandmeier** [comment]
Any update here?
I'm running Godot 3.3.3 stable official and this bug still persists.

![image](https://user-images.githubusercontent.com/26724996/136311932-a080b4f0-0657-46ed-9035-f5920052e9e3.png)

Top (incorrect) happens after setting the bitmask for the first time. Bottom (correct) is after exiting and restarting Godot.
Here's the bitmask:

![image](https://user-images.githubusercontent.com/26724996/136312042-84d023ec-77b2-4bc9-a47e-60579b0937e1.png)


I hope this helps.

**Comment by klynicol** [comment]
Yeah, same issue happened to me. Just spent over and hour thinking I was going nuts.
